### PR TITLE
Reuse the aspect used by node_binary/node_test to gather upstreams sources to gather the manifest required by jasmine.

### PR DESCRIPTION
**Pull request by LucasSloan**

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: internal/devmode_js_sources.bzl [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
   ]))
   return [DefaultInfo(files = files)]
 
 devmode_js_sources = rule(
     implementation = _devmode_js_sources_impl,
     attrs = {
-        "deps": attr.label_list(allow_files = True),
+        "deps": attr.label_list(
+            allow_files = True,
+            aspects=[sources_aspect],
```
Comment (alexeagle) [inline internal/devmode_js_sources.bzl]: nit: spaces around `=` (really we need a linter/formatter for .bzl)

Comment (LucasSloan) [reply internal/devmode_js_sources.bzl]: Fixed.

File: examples/program/BUILD.bazel [hunk 2]
```diff
@@ -12,15 +13,20 @@ filegroup(
     visibility = ["//visibility:public"],
 )
 
+mock_typescript_lib(
+    name = "tots_typescript",
```
Comment (alexeagle) [inline examples/program/BUILD.bazel]: cute name, but let's just be descriptive, like "mock_ts_lib" orsomething

File: internal/typescript_mock.bzl [hunk 3]
```diff
@@ -0,0 +1,12 @@
+def _mock_typescript_lib(ctx):
```
Comment (alexeagle) [inline internal/typescript_mock.bzl]: give a quick doc, like "allows testing that node_jasmine_test will work with ts_library from rules_typescript without introducing a circular dependency"

File: examples/program/BUILD.bazel (hunk 2 above)
Comment (LucasSloan) [reply examples/program/BUILD.bazel]: Done.

**Pull request merged**